{"ast":null,"code":"var __awaiter = this && this.__awaiter || function (thisArg, _arguments, P, generator) {\n  function adopt(value) {\n    return value instanceof P ? value : new P(function (resolve) {\n      resolve(value);\n    });\n  }\n  return new (P || (P = Promise))(function (resolve, reject) {\n    function fulfilled(value) {\n      try {\n        step(generator.next(value));\n      } catch (e) {\n        reject(e);\n      }\n    }\n    function rejected(value) {\n      try {\n        step(generator[\"throw\"](value));\n      } catch (e) {\n        reject(e);\n      }\n    }\n    function step(result) {\n      result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected);\n    }\n    step((generator = generator.apply(thisArg, _arguments || [])).next());\n  });\n};\nexport function onErrorify(originalMethod, config) {\n  return function (...args) {\n    return __awaiter(this, void 0, void 0, function* () {\n      const onErrorFunc = typeof config.func === 'string' ? this[config.func].bind(this) : config.func;\n      try {\n        const res = yield originalMethod.apply(this, args);\n        return res;\n      } catch (e) {\n        return onErrorFunc(e, args);\n      }\n    });\n  };\n}\n//# sourceMappingURL=on-errorify.js.map","map":null,"metadata":{},"sourceType":"module","externalDependencies":[]}